src/shared/api/env/env.ts

```diff
@@ -1,13 +1,6 @@
-import { useQuery } from '@tanstack/react-query';
 import { ClientEnv } from '@/shared/api/env/types';
 
-export const useEnv = () => {
-  return useQuery({
-    queryKey: ['clientEnv'],
-    queryFn: async (): Promise<ClientEnv> => {
-      const res = await fetch('/api/env');
-      return (await res.json()) as ClientEnv;
-    },
-    staleTime: Infinity,
-  });
```

Comment:
Moving away a bit from composing too much with hooks. QueryFn's that depend on env can just do an async await for it. While it may seem like multiple fetches will occur and we don't take advantage of react-query caching, it actually gets cached just fine with http caching in-browser.